### PR TITLE
[JUJU-1512] Update juju/cmd to v3.0.1

### PR DESCRIPTION
**Pull request by barrettj12**
This pulls through a small change (juju/cmd#81) so now we don't add an extra newline to output with non-default formatters.

## Changes

The following commands have had an extraneous newline removed after tabular output. I have changed the tests to match.
- `exec`
- `operations`
- `run`
- `show-task`
- `disabled-commands`
- `clouds`
- `credentials`
- `regions`
- `enable-ha`
- `controllers` (+ feature tests)
- `models` (+ feature tests)
- `find-offers`
- `offers`
- `show-offer`
- `list-firewall-rules`
- `machines`
- `show-machine`
- `model-config`
- `charm-resources`
- `resources` (+ feature tests)
- `secrets`
- `show-status-log`
- **`status`** (+ feature tests)
  - The `status --format oneline` output actually had an extra newline **before**, which I have removed.
- `storage` (+ feature tests)
- `storage-pools` (+ feature tests)
- `users` (+ feature tests)
- `model-defaults` (feature tests only)
- `jujuc secret-get` for yaml output

The following commands didn't have an extraneous newline, so I've updated their formatting code to explicitly add a newline:
- `metrics`
- `commits`
- `payloads`
- `agreements`
- `plans`
- `wallets`
- `show-wallet`
- `sla`
- `space`
- `move-to-space`
- `whoami`
- `get-constraints`
- `get-model-constraints`
- `jujuc secret-get` for single output

## TODO
- [x] fix the status command/tests
- [ ] check the CI passes
- [x] test a few commands manually

### CODE REVIEW

File: cmd/juju/metricsdebug/metrics.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func formatTabular(writer io.Writer, value interface{}) error {
 	for _, m := range metrics {
 		table.AddRow(m.Unit, m.Timestamp.Format(time.RFC3339), m.Metric, m.Value, renderLabels(m.Labels))
 	}
-	_, err := fmt.Fprint(writer, table.String())
+	_, err := fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/metricsdebug/metrics.go:176]: Fprintln?

File: cmd/juju/model/listcommits.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ func (c *CommitsCommand) printTabular(writer io.Writer, value interface{}) error
 	for _, c := range list.Commits {
 		table.AddRow(c.CommitId, c.CommittedAt, c.CommittedBy, c.BranchName)
 	}
-	_, _ = fmt.Fprint(writer, table)
+	_, _ = fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/model/listcommits.go:172]: Fprintln?

File: cmd/juju/romulus/listagreements/listagreements.go [hunk 3]
```diff
@@ -137,7 +137,7 @@ func formatTabular(writer io.Writer, value interface{}) error {
 		}
 	}
 
-	_, err := fmt.Fprint(writer, table)
+	_, err := fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/romulus/listagreements/listagreements.go:140]: Fprintln?

File: cmd/juju/romulus/listplans/list_plans.go [hunk 4]
```diff
@@ -205,7 +205,7 @@ func formatTabular(writer io.Writer, value interface{}) error {
 	for _, plan := range plans {
 		table.AddRow(plan.URL, plan.Price, plan.Description)
 	}
-	fmt.Fprint(writer, table)
+	fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/romulus/listplans/list_plans.go:208]: Fprintln?

File: cmd/juju/romulus/listwallets/list-wallets.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ func formatTabular(writer io.Writer, value interface{}) error {
 	table.AddRow("Total", b.Total.Limit, b.Total.Budgeted, b.Total.Available, b.Total.Consumed)
 	table.AddRow("", "", "", "", "")
 	table.AddRow("Credit limit:", b.Credit, "", "", "")
-	fmt.Fprint(writer, table)
+	fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/romulus/listwallets/list-wallets.go:115]: Fprintln?

File: cmd/juju/romulus/showwallet/show_wallet.go [hunk 6]
```diff
@@ -125,7 +125,7 @@ func (c *showWalletCommand) formatTabular(writer io.Writer, value interface{}) e
 	table.AddRow("Total", b.Total.Consumed, b.Total.Budgeted, "", b.Total.Usage)
 	table.AddRow("Wallet", "", b.Limit, "")
 	table.AddRow("Unallocated", "", b.Total.Unallocated, "")
-	fmt.Fprint(writer, table)
+	fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/romulus/showwallet/show_wallet.go:128]: Fprintln?

File: cmd/juju/romulus/sla/sla.go [hunk 7]
```diff
@@ -226,7 +226,7 @@ func formatTabular(writer io.Writer, value interface{}) error {
 	}
 	table.AddRow("Model", "SLA", "Message")
 	table.AddRow(l.Model, l.SLA, l.Message)
-	fmt.Fprint(writer, table)
+	fmt.Fprint(writer, table.String()+"\n")
```
Comment (hpidcock) [inline cmd/juju/romulus/sla/sla.go:229]: Fprintln... you get the idea

File: cmd/juju/status/status_internal_test.go [hunk 8]
```diff
@@ -5040,7 +5037,7 @@ func (s *StatusSuite) TestStatusWithFormatOneline(c *gc.C) {
 - wordpress/0: 10.0.1.1 (agent:idle, workload:active)
   - logging/0: 10.0.1.1 (agent:idle, workload:active)
 `
```
Comment (hpidcock) [inline cmd/juju/status/status_internal_test.go:5039]: generally I'd put the `[1:]` at the end of the string literal

File: worker/uniter/runner/jujuc/secret-get.go [hunk 9]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/juju/cmd/v3"
 	"github.com/juju/errors"
 	"github.com/juju/gnuflag"
+
 	"github.com/juju/juju/core/secrets"
```
Comment (hpidcock) [inline worker/uniter/runner/jujuc/secret-get.go:15]: remove new line

**Review by hpidcock — APPROVED** [review]

File: cmd/juju/firewall/listrules_test.go [hunk 10]
```diff
@@ -9,10 +9,11 @@ import (
 	"github.com/juju/cmd/v3"
 	"github.com/juju/cmd/v3/cmdtesting"
 	"github.com/juju/errors"
-	"github.com/juju/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
+	"github.com/juju/juju/testing"
+
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/firewall/listrules_test.go:16]: Bonkers that `goimports` can't do this for me. Might look into https://github.com/rinchsan/gosimports

**Pull request merged**